### PR TITLE
pgcryptocipherccl: convert invariant errors into assertions

### PR DESCRIPTION
**Pull request by andyyang890**
This patch changes the checks for programmer error in the `pgcryptocipherccl` package to use `errors.AssertionFailedf` so that they are surfaced with greater visibility.

Epic: None

Release note: None

### CODE REVIEW

File: pkg/ccl/pgcryptoccl/pgcryptocipherccl/cipher.go [hunk 1]
```diff
@@ -155,6 +155,6 @@ func decrypt(method cipherMethod, block cipher.Block, iv []byte, data []byte) ([
 		mode.CryptBlocks(ret, data)
 		return ret, nil
 	default:
-		return nil, errors.Newf("cannot encrypt for unknown mode: %d", m)
+		return nil, errors.AssertionFailedf("cannot encrypt for unknown mode: %d", m)
```
Comment (rafiss) [inline pkg/ccl/pgcryptoccl/pgcryptocipherccl/cipher.go:158]: nit: this should say `decrypt`

**Review by rafiss — APPROVED** [review]
lgtm, but just one nit! just to check - none of these errors can occur from strange user input, right?

**Review by andyyang890 — COMMENTED** [review]
> just to check - none of these errors can occur from strange user input, right?

Yup! The assertions in `cipher.go` should only be hit if we forgot a enum case in the switch statement and the assertion in `pkcsPad` would only happen if we passed in an invalid block size (not directly configurable by user). Neither of these is currently happening.

**Comment by andyyang890** [comment]
tftr!

bors r=rafiss

**Pull request merged**